### PR TITLE
Bump to use vfs4

### PR DESCRIPTION
**Pull request by Itxaka**
Requires the latest yip version as that one has been bumped to vfs5 as well

### CODE REVIEW

**Review by mauromorales — APPROVED** [review]

File: pkg/utils/chroot.go [hunk 1]
```diff
@@ -149,12 +148,11 @@ func (c *Chroot) RunCallback(callback func() error) (err error) {
 	}()
 
 	// Store current root
-	oldRootF, err = c.config.Fs.Open("/")
+	rootPath, _ := c.config.Fs.RawPath("/")
```
Comment (Itxaka) [inline pkg/utils/chroot.go:151]: we no longer need to open the oldRoot directly to be able to chdir to it later, which is nice

File: pkg/utils/common.go [hunk 2]
```diff
@@ -163,6 +163,14 @@ func SyncData(log sdkTypes.KairosLogger, runner v1.Runner, fs v1.FS, source stri
 		}
 		if t, err := fs.RawPath(target); err == nil {
 			target = t
+			// create target path if it doesnt exists
+			if _, err := os.Stat(target); err != nil {
```
Comment (Itxaka) [inline pkg/utils/common.go:167]: looks like older vfs was more strict when dir were not there. Now its more relaxed and it doesnt fail so we need to check ourselves.

File: pkg/utils/common.go [hunk 3]
```diff
@@ -451,7 +463,8 @@ func FindFileWithPrefix(fs v1.FS, path string, prefixes ...string) (string, erro
 		}
 		for _, p := range prefixes {
 			if strings.HasPrefix(f.Name(), p) {
-				if f.Mode()&os.ModeSymlink == os.ModeSymlink {
+				info, _ := f.Info()
+				if info.Mode()&os.ModeSymlink == os.ModeSymlink {
```
Comment (Itxaka) [inline pkg/utils/common.go:467]: now we dont have the mode directly so we need to info it

File: pkg/utils/fs/fs.go [hunk 4]
```diff
@@ -236,12 +236,7 @@ func walkDir(fs v1.FS, path string, d fs.DirEntry, walkDirFn fs.WalkDirFunc) err
 }
 
 func readDir(fs v1.FS, dirname string) ([]fs.DirEntry, error) {
-	f, err := fs.Open(dirname)
-	if err != nil {
-		return nil, err
-	}
-	dirs, err := f.ReadDir(-1)
-	f.Close()
+	dirs, err := fs.ReadDir(dirname)
```
Comment (Itxaka) [inline pkg/utils/fs/fs.go:239]: now it directly opens the dir, this function is mainly obsolete as it does the same as the underlying!

**Review by mudler — APPROVED** [review]
looking good here!

**Pull request merged**